tests/index.test.ts

```diff
@@ -187,169 +233,157 @@ describe('ParaSwap SDK', () => {
 
   test('Build_Tx', async () => {
     const destToken = '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48';
-    const ratesOrError = await paraSwap.getRate(
+    const priceRoute = await paraSwap.getRate({
       srcToken,
       destToken,
-      srcAmount,
-      senderAddress,
-      SwapSide.SELL,
-      { includeDEXS: 'UniswapV2' }
-    );
-
-    expect((ratesOrError as APIError)?.data?.error).toBeUndefined();
-
-    const priceRoute = ratesOrError as OptimalRate;
+      amount: srcAmount,
+      userAddress: senderAddress,
+      side: SwapSide.SELL,
+      options: {
+        includeDEXS: 'UniswapV2',
+      },
+    });
 
     const destAmount = new BigNumber(priceRoute.destAmount)
       .times(0.99)
       .toFixed(0);
 
-    const txOrError = await paraSwap.buildTx(
-      srcToken,
-      destToken,
-      srcAmount,
-      destAmount,
-      priceRoute,
-      senderAddress,
-      referrer,
-      undefined,
-      undefined,
-      undefined,
-      { ignoreChecks: true }
-    );
-
-    expect(txOrError.data.error).toBeUndefined();
-    expect(typeof txOrError).toBe('object');
-  });
-  if (TESTING_ENV) {
-    test('Build_tx_legacy', async () => {
-      const ratesOrError = await paraSwap.getRate(
-        srcToken,
-        destToken,
-        srcAmount,
-        senderAddress,
-        SwapSide.SELL,
-        { includeDEXS: 'Uniswap,UniswapV2,Balancer,Oasis' }
-      );
-      const priceRoute = ratesOrError as OptimalRate;
-      const destAmount = new BigNumber(priceRoute.destAmount)
-        .times(0.99)
-        .toFixed();
-
-      const txOrError = await paraSwap.buildTx(
+    const txParams = await paraSwap.buildTx(
+      {
         srcToken,
         destToken,
         srcAmount,
         destAmount,
         priceRoute,
-        signer.address,
-        referrer,
-        undefined,
-        undefined,
-        undefined,
-        { ignoreChecks: true }
-      );
-      const transaction = txOrError as TransactionParams;
-      expect(typeof transaction).toBe('object');
+        userAddress: senderAddress,
+        partner: referrer,
+      },
+      { ignoreChecks: true }
+    );
+
+    expect(typeof txParams).toBe('object');
```

Comment:
It would be interesting to use `toMatchObject` in such cases
Though it's a chore writing out the object schema.

Reply:
sounds good, will try to adjust